opentracing-cdi/src/main/java/io/opentracing/contrib/cdi/internal/SpanContextProducer.java

```diff
@@ -0,0 +1,47 @@
+package io.opentracing.contrib.cdi.internal;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.util.logging.Logger;
+
+/**
+ * Produces a {@link SpanContext}, based on the {@link HttpServletRequest}'s attributes.
+ */
+public class SpanContextProducer {
+    private static final Logger log = Logger.getLogger(SpanContextProducer.class.getName());
+
+    @Inject
+    HttpServletRequest request;
+
+    @Inject
+    Tracer tracer;
+
+    @Produces
+    public SpanContext produceSpanContext() {
+        SpanContext spanContext = (SpanContext) request.getAttribute(TracingFilter.SERVER_SPAN_CONTEXT);
```

Comment:
I don't have a strong opinion about this but I think ActiveSpan producer might be enough. These to represent the same span. 

minor not in async request `request.getAttribute(TracingFilter.SERVER_SPAN_CONTEXT);` would return null and I guess active span producer too.